### DISCUSSION TITLE
Another H3 Pro

### DISCUSSION THREAD

**Discussion by a900a900a900** [nathanmarlor/foxess_modbus#684] [Support] [answered]
Connection Status (Fox ESS Inverter) changed to Error reading registers. Type: RegisterType.HOLDING; start: 36001; count: 3; slave: 247 from tcp://192.168.4.150:502: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)

Its reporting the inverter as a H3 Pro and showing entity's in HA. But not seeing values.

Latest software as far as I can. Serial port data on the web server shows data back and forth and no failed bytes. Connections made to EMS rs485 ports. Connection via EW11

**Comment by FozzieUK** [comment]
Hi, the error message is saying that the no response was received - has it ever worked, or are you still trying to get it working?

As there is no response, I think the best thing is to double check the basics - first that you are wired to RS485A on pin17 and RS485B on pin 18 of the H3P's 20 way Meter/CT connector.

And that the wires are connected straight through - so 485A (pin17) goes to the RS485A pin on the Elfin and 485B (pin 18) goes to 485B on the Elfin.

On the H3P can you make sure that the slave ID has been set to 247 (In Setting, Communication, RS485)

At this point, probably worth trying to connect again and see what the response is.

If that doesn't work, I think you'll need to go through the EW11 setup and double checking each of the parameters, paying special attention to the protocols and baud rate.

**Reply by a900a900a900** [reply]
Never worked. Trying to get it working. Checked the wiring. Pins 17 & 18 EMS pins to the outer 2 on the EW11.
Communication/RS485/DEVICEID-2/247.

Checked the EW webpage and all settings match the guide

**Reply by a900a900a900** [reply]
Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)

I know that would appear pretty obvious that ive not connected something correctly

**Reply by a900a900a900** [reply]
Error reading registers. Type: RegisterType.HOLDING; start: 37609; count: 92; slave: 247 from tcp://192.168.0.150:502: Exception Response(131, 3, IllegalAddress)

Switched the RS485 addresses in the inverter and get this

**Reply by FozzieUK** [reply, marked as answer]
That looks better, at least it’s getting a response now - try this as discussed in this recent post https://github.com/nathanmarlor/foxess_modbus/discussions/685#discussioncomment-10810272

**Reply by FozzieUK** [reply]
Ok great, this appears to be an H3 Pro issue as it has so many registers compared to a single phase hybrid, you'll get away with a higher number than that, probably as high as 50 - it won't make a big difference in speed on the Elfin as long as it's reliable.

**Reply by a900a900a900** [reply]
changing registers to 10 has caused all sensors to show, Those that can be changed seem to respond to changes too.

**Reply by canton7** [reply]
I'm guessing we should probably reduce the max to 50 across the board?

**Reply by FozzieUK** [reply]
Yep I think that’s the safest option 👍
